log4j-api/src/main/java/org/apache/logging/log4j/spi/AbstractLogger.java

```diff
@@ -2742,11 +2739,8 @@ public LogBuilder atFatal() {
      */
     @Override
     public LogBuilder always() {
-        final DefaultLogBuilder builder = logBuilder.get();
-        if (builder.isInUse()) {
-            return new DefaultLogBuilder(this);
-        }
-        return builder.reset(Level.OFF);
+        final DefaultLogBuilder builder = (DefaultLogBuilder) recycler.acquire();
+        return builder.atLevel(Level.OFF);
```

Comment:
...so that we don't need this unchecked cast.

I can open a new issue for this, since this PR is getting huge.